# Python Exception Handling

## Introduction

This directory contains programs and functions that demonstrate
python's error and exception handling abilities.

## 0-safePprint_list.py

A function that prints a specified number of elements from a list on the same
line followed by a new line. It also handles any errors that may arise from
attempting to print an out-of-range element using a try, except block.

## 1-safe_print_integer.py

A function that prints given integer input in a specified format and reports
when non-integer input is provided. Handles any errors that arise from
attempting to print non-integer input using the specified format.

## 2-safe_print_list_integers.py

A function that prints a specified number of elements from a list that are only
integers. Other non-integer elements in the list are silently skipped.
The funtion handles any error that may arise from trying to print an
out-of-range integer element.

## 3-safe_print_division.py

A function that takes in two integers and prints the result of their division.
It also handles any errors that may arise from trying to divide by zero.

## 4-list_division.py

A function that divides the elements of one list by the corresponding elements
of another list and generates a new list that contains the result of each
division operation. The function handles any errors arising from either one of
the elements not being an integer or from a division by zero error.

## 5-raise_exception.py

A function that raises a TypeError exception.

## 6-raise_exception_msg.py

A function that raises a name exception with a message.